client/src/components/Survey.tsx

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import React from 'react';
+import { Link, Stack } from '@fluentui/react';
+import { PostCallConfig } from '../models/ConfigModel';
+import { rejoinLinkStyle, surveyStyle } from '../styles/Survey.styles';
+
+export interface SurveyProps {
+  onRejoinCall: () => void;
+  postCall?: PostCallConfig;
```

Comment:
Since we only want to render the survey when postCall actually exists, then I think it is better we make this non-nullable.

Reply:
Since we are flowing down the config from AppConfigModel->Visit.tsx->MeetingExperience.tsx->Survey.tsx , I declared it as an optional field. Making this mandatory here throws compile errors in places where it is being assigned . 

AppConfigModel has postCall as optional and so in other components using it too, it's marked optional. 

I am making the change to have the if condition checking the postCallType before the subscription to onCallEnded event on adapter and am in the process of testing it. That will make sure that Survey is rendered only in the postcall scenarios. What do you think?

Reply:
It is annoying to have all the nullability checks (`props.postCall?.survey?.options.surveyUrl`) in the `Survey` component. Instead could `Survey`'s props interface demand that `postCall` be non-nullable? Then whoever calls the `Survey` needs to provide proof that it has non-null `postCall`:

```typescript
if (renderPostCall && postCall) {
  return (
    <Survey postCall={postCall} ... />
  );
}
```

Reply:
Thanks! Have made this change.